odltools/mdsal/models/network_topology.py

```diff
@@ -50,8 +50,38 @@ def get_nodes_by_tid_and_key(self, tid="ovsdb:1", key='node-id'):
             d[node[key]] = node
         return d
 
+    def get_nodes_by_dpid(self, tid="ovsdb:1"):
+        d = {}
+        nodes = self.get_nodes_by_tid(tid)
+        for node in nodes:
+            dpid = self.get_datapathid_from_node(node)
+            if dpid:
+                d[dpid] = node
+        return d
+
+    def get_dpn_host_mapping(self):
+        d = {}
+        nodes = self.get_nodes_by_tid_and_key()
+        for nodeid in nodes:
+            node = nodes.get(nodeid)
+            dpid = self.get_datapathid_from_node(node)
+            if dpid:
+                parentid = self.get_parent_nodeid(nodeid)
+                host = self.get_host_id_from_node(nodes.get(parentid))
+                d[dpid] = host
+        return d
+
     def get_host_id_from_node(self, node):
         extids = node.get("ovsdb:openvswitch-external-ids", {})
         for extid in extids:
-            if extid.get("external-id-key") == "odl_os_hostconfig_hostid":
+            if extid.get("external-id-key") == "hostname":
```

Comment:
why not leave this method as is and create a new one: get_hostname_from_node().

with this patch, does this mean only hostname field is used? That was coming back as localhost in the csit so it wasn't accurate.